skynet/modules/ttt/openai_api/app.py

```diff
@@ -9,37 +12,76 @@
     llama_path,
     openai_api_base_url,
     openai_api_server_port,
-    vllm_server_path,
 )
 from skynet.logs import get_logger
-from skynet.utils import get_device
+from skynet.utils import create_app, dependencies, get_device, responses
 
-proc = None
 use_vllm = get_device() == 'cuda'
 
 log = get_logger(__name__)
 
 
-def initialize():
-    log.info('Starting OpenAI API server...')
+async def run_vllm_server(args, main_app: FastAPI):
+    from vllm.entrypoints.launcher import serve_http
+    from vllm.entrypoints.openai.api_server import (
+        build_async_engine_client,
+        init_app_state,
+        lifespan,
+        router,
+        TIMEOUT_KEEP_ALIVE,
+    )
+
+    async with build_async_engine_client(args) as engine_client:
+        app = create_app(lifespan=lifespan)
```

Comment:
Looks like creating the app itself doesn't require any vllm async stuff, does it? If so we could reverse the logic, which is IMHO cleaner.

That is, create the app like we do others, then mount it in the main file, and pass it for further initialization, WDYT?

Reply:
wanted to centralize the checks for whether vllm or llama need to be loaded